api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupActiveOnCareList.java

```diff
@@ -0,0 +1,159 @@
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding
+     * properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "ActiveoncareList.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:7,dataset:ACTIVEONCARE_LIST");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.setParameters(getParameters());
+
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+
+        CohortDefinition hadEncounterInQuarter = hivCohortDefinitionLibrary.getArtPatientsWithEncounterOrSummaryPagesBetweenDates();
+
+        CohortDefinition returnVisitDuringPeriod = df.getPatientsWhoseObsValueDateIsBetweenStartDateAndEndDate(hivMetadata.getReturnVisitDate(),
+                Arrays.asList(hivMetadata.getARTEncounterEncounterType()), BaseObsCohortDefinition.TimeModifier.ANY);
+        CohortDefinition longAppointments = df.getPatientsWithLongRefills();
+        CohortDefinition definition = df.getPatientsInAny(returnVisitDuringPeriod,longAppointments,hadEncounterInQuarter);
+
+        dsd.setName(getName());
+        dsd.setParameters(getParameters());
+        dsd.addRowFilter(Mapped.mapStraightThrough(definition));
+        addColumn(dsd, "Clinic No", hivPatientData.getClinicNumber());
+        addColumn(dsd, "Family Name", builtInPatientData.getPreferredFamilyName());
+        addColumn(dsd, "Given Name", builtInPatientData.getPreferredGivenName());
+        addColumn(dsd, "Sex", builtInPatientData.getGender());
+        dsd.addColumn("Birth Date", builtInPatientData.getBirthdate(), "", new BirthDateConverter());
+        addColumn(dsd,"Age",hivPatientData.getAgeDuringPeriod());
+        addColumn(dsd, "Telephone", basePatientData.getTelephone());
+        addColumn(dsd,"Address",basePatientData.getVillage());
+        addColumn(dsd,"EnrollmentDate",hivPatientData.getEnrollmentDate());
+        addColumn(dsd,"ARTStartDate",hivPatientData.getArtStartDate());
+        addColumn(dsd,"BaseLineCD4",hivPatientData.getBaselineCD4());
+        addColumn(dsd,"CurrentRegimen",hivPatientData.getCurrentRegimen());
+
+        addColumn(dsd,"lastEncounterDate",hivPatientData.getLastEncounterDuringPeriod());
```

Comment:
The encounter does not have to be in the period it could be before the period but not after